Doc/library/sqlite3.rst

```diff
@@ -47,83 +47,104 @@ This document includes four main sections:
       PEP written by Marc-André Lemburg.
 
 
+.. We use the following practises for SQL code:
+.. - UPPERCASE for keywords
+.. - lowercase for schema
+.. - single quotes for string literals
+.. - singular for table names
+.. - if needed, use double quotes for table and column names
+
 .. _sqlite3-tutorial:
 
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating an on-disk database :file:`python.db`
+for managing Python release metadata.
+A fundamental understanding of database concepts,
+like `transactions`_ and `cursors`_, is assumed.
+
+To start working with an SQLite database,
+you must first open a connection to it.
+Call the :func:`sqlite3.connect` function
+to create a :class:`Connection` to the database :file:`python.db`::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("python.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+SQLite will implicitly create the database file :file:`python.db`
+in the current working directory if it does not exist.
+The returned :class:`Connection` object ``con``
+represents our connection to the on-disk database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+we need to use a database cursor.
+Call the :meth:`~Connection.cursor` method of ``con`` to create
+:class:`Cursor` ``cur``::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
-
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
-
-   # Save (commit) the changes
-   con.commit()
+Now, let us create a database table ``release`` with columns for version
+number and release date, and inserting the release data for Python 1.0.
+Execute the SQL statements
+by calling the :meth:`~Cursor.execute` method of ``cur``::
 
-   # We can also close the connection if we are done with it.
-   # Just be sure any changes have been committed or they will be lost.
-   con.close()
+   cur.execute("""CREATE TABLE release(version, releasedate)""")
+   cur.execute("INSERT INTO release VALUES('3.0', '2008-12-03')")
 
-The saved data is persistent: it can be reloaded in a subsequent session even
-after restarting the Python interpreter::
+The ``INSERT`` statement implicitly opens a transaction
+that needs to be committed before changes are saved in the database
+(see :ref:`sqlite3-controlling-transactions` for details).
+Call the :meth:`~Connection.commit` on the connection object
```

Comment:
Typo; this should be either

```suggestion
Call the :meth:`~Connection.commit` method on the connection object
```

or

```suggestion
Call :meth:`~Connection.commit` on the connection object
```

Reply:
I prefer the latter; it is concise. I'll make that change throughout the tutorial.